### PR TITLE
[rushjs.io] Document "rush-pnpm" and "rushx" and Rush's folder tree

### PR DESCRIPTION
**Pull request by octogonz**
This also includes docs for the https://github.com/microsoft/rushstack/pull/3872 feature

### CODE REVIEW

File: websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+| common/temp/node_modules/...  | The installed packages. This is a plain old `npm install` output, with no symlinks in this tree.                                             |
+| common/temp/npm-cache/...     | A local NPM cache will be created here. Rush does not use the global NPM cache due to its concurrency problems.                              |
+| common/temp/npm-local/...     | Based on the **npmVersion** setting, Rush will install NPM in your home directory and create a symlink to it for each repo that requests it. |
+| common/temp/npm-tmp/...       | Temporary files created by NPM during installation.                                                                                          |
+| common/temp/projects/...      | Synthetic projects referenced by **common/temp/package.json**.                                                                               |
+| common/temp/rush-recycler/... | Used to speed up recursive deletes.                                                                                                          |
```
Comment (stekycz) [inline websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md:51]: Would it make sense to mention `common/temp/telemetry/...` directory here?

File: websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+| common/temp/build-cache/...   | Default storage location for Rush's [Build cache](../maintainer/build_cache.md)                                                              |
+| common/temp/install-run/...   | Storage for the **install-run.js** and **install-run-rush.js** scripts. See [Enabling CI builds](../maintainer/enabling_ci_builds.md).       |
+| common/temp/node_modules/...  | The installed packages. This is a plain old `npm install` output, with no symlinks in this tree.                                             |
+| common/temp/npm-cache/...     | A local NPM cache will be created here. Rush does not use the global NPM cache due to its concurrency problems.                              |
+| common/temp/npm-local/...     | Based on the **npmVersion** setting, Rush will install NPM in your home directory and create a symlink to it for each repo that requests it. |
+| common/temp/npm-tmp/...       | Temporary files created by NPM during installation.                                                                                          |
```
Comment (stekycz) [inline websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md:49]: Would it make sense to mention `common/temp/pnpm-local/...` and `common/temp/pnpm-store/...` as well? What about Yarn specific directories?

File: websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+| common/scripts/install-run-rushx.js     | CI bootstrap script for invoking [rushx](../commands/rushx.md).                                                                                                                              |
+| common/scripts/install-run.js           | CI bootstrap script for invoking arbitrary NPM packages.                                                                                                                                     |
+
+## Temporary files created by Rush
+
+| Folder path                   | What it does                                                                                                                                 |
+| :---------------------------- | :------------------------------------------------------------------------------------------------------------------------------------------- |
+| common/temp/build-cache/...   | Default storage location for Rush's [Build cache](../maintainer/build_cache.md)                                                              |
```
Comment (stekycz) [inline websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md:44]: Would it make sense to mention `common/deploy/...` directory here?

File: websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+| Folder path                             | What it does                                                                                                                                                                                 |
+| :-------------------------------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| common/autoinstallers/...               | [Autoinstaller projects](../maintainer/autoinstallers.md) are created under this folder                                                                                                      |
+| common/changes/...                      | Stores change files created by the [rush change](../commands/rush_change.md) command and consumed by the [rush version](../commands/rush_version.md) command.                                |
+| common/git-hooks/...                    | Rush's [git hook scripts](../maintainer/git_hooks.md) are defined here                                                                                                                       |
```
Comment (stekycz) [inline websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md:34]: Would it make sense to mention `common/pnpm-patches/...` directory here?

File: websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+| common/temp/install-run/...   | Storage for the **install-run.js** and **install-run-rush.js** scripts. See [Enabling CI builds](../maintainer/enabling_ci_builds.md).       |
+| common/temp/node_modules/...  | The installed packages. This is a plain old `npm install` output, with no symlinks in this tree.                                             |
+| common/temp/npm-cache/...     | A local NPM cache will be created here. Rush does not use the global NPM cache due to its concurrency problems.                              |
+| common/temp/npm-local/...     | Based on the **npmVersion** setting, Rush will install NPM in your home directory and create a symlink to it for each repo that requests it. |
+| common/temp/npm-tmp/...       | Temporary files created by NPM during installation.                                                                                          |
+| common/temp/projects/...      | Synthetic projects referenced by **common/temp/package.json**.                                                                               |
```
Comment (stekycz) [inline websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md:50]: Would it make sense to mention `common/temp/patches/...` directory here?

File: websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+| common/config/rush/npm-shrinkwrap.json                                          | The shrinkwrap file when your package manager is NPM. This is the common shrinkwrap file that applies to all projects in the Rush repo. For more information, see **"What is this "shrinkwrap file"** in the [Everyday commands](../developer/everyday_commands.md) section. |
+| common/config/rush/pnpm-lock.yaml                                               | The shrinkwrap file when your package manager is PNPM.                                                                                                                                                                                                                       |
+| common/config/rush/yarn.lock                                                    | The shrinkwrap file when your package manager is Yarn.                                                                                                                                                                                                                       |
```
Comment (stekycz) [inline websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md:22]: Would it make sense to mention `common/config/rush/repo-state.json` file here?

File: websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+| common/config/rush/nonbrowser-approved-packages.json                            | Used by the **approvedPackagesPolicy** setting from rush.json                                                                                                                                                                                                                |
+| [common/config/rush/pnpm-config.json](../configs/pnpm-config_json.md)           | Configuration specific to the PNPM package manager                                                                                                                                                                                                                           |
+| [common/config/rush/version-policies.json](../configs/version-policies_json.md) | Defines the [rush version](../commands/rush_version.md) and [rush publish](../commands/rush_publish.md) workflows.                                                                                                                                                           |
```
Comment (stekycz) [inline websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md:26]: Would it make sense to mention `common/config/rush/rush-plugins.json` file here?

**Review by stekycz — COMMENTED** [review]
Thank you for updating the documentation. I just left a few suggestions regarding (probably) forgotten files and directories.

File: websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md (hunk 1 above)
Comment (octogonz) [reply websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md:51]: Yes, good suggestion!

File: websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md (hunk 2 above)
Comment (octogonz) [reply websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md:49]: Added 👍

File: websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md (hunk 3 above)
Comment (octogonz) [reply websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md:44]: I added it in the `Configuration files` section

File: websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md (hunk 4 above)
Comment (octogonz) [reply websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md:34]: Added

File: websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md (hunk 5 above)
Comment (octogonz) [reply websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md:50]: I wasn't aware of this one -- had to read about how `rush-pnpm commit-patch` works 😆

File: websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md (hunk 6 above)
Comment (octogonz) [reply websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md:22]: Added

File: websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md (hunk 7 above)
Comment (octogonz) [reply websites/rushjs.io/docs/pages/advanced/rush_files_and_folders.md:27]: Added

**Pull request merged**